### ISSUE TITLE
🚀 Feature Request: Enable caching for assets binding

### ISSUE THREAD

**Issue by rdlogout** [cloudflare/workers-sdk#7650]
### Describe the solution

Hi,

I have been using cloudflare pages for my frontend.
Recently i tried to migrate it to worker new asset binding.

But i noticed the site become little slow, when i looked for the cause i found the CF Pages app respect _header and return response with cache 
![image](https://github.com/user-attachments/assets/7b116027-52f8-482d-b006-37f8bd7a4a78)
but when i tested a skelton app with worker.dev asset binding it return
![image](https://github.com/user-attachments/assets/13ea2432-23d0-4352-85c7-dca45cbaa868)

Can u allow _header file in worker asset binding too, it will solve the performance issue.

Thank you

**Comment by rdlogout** [comment]
![image](https://github.com/user-attachments/assets/56e83782-57e3-425c-8c85-a9b80a5306f9)
https://github.com/cloudflare/workers-sdk/blob/f785f128b6530b61d2a9e59e7bbeb22189a5b722/packages/workers-shared/asset-worker/src/constants.ts

**Comment by WillTaylorDev** [comment]
(As an aside, and I know it's not what you're asking for here-- you _could_ use a `experimental_serve_directly=false` along with a worker to dynamically fetch your assets. You then could modify the cache-control header yourself before sending the response back. The downside to this is that worker invocations count towards chargeable requests (but might work for smaller sites)).

**Comment by rdlogout** [comment]
> (As an aside, and I know it's not what you're asking for here-- you _could_ use a `experimental_serve_directly=false` along with a worker to dynamically fetch your assets. You then could modify the cache-control header yourself before sending the response back. The downside to this is that worker invocations count towards chargeable requests (but might work for smaller sites)).

Yes i tried that but as u mentioned it count as invocation and its not ideal for my use case where we have thousands js chunks

**Comment by petebacondarwin** [comment]
Cache in front of assets is being worked on...
